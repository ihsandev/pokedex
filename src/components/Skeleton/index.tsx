import Line from "./line";

const Skeleton = {
  Line: Line
};

export default Skeleton;
