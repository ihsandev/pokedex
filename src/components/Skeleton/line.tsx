import React from "react";
import styled, { keyframes } from "styled-components";

export interface Props extends React.HTMLAttributes<HTMLDivElement> {
  width?: string | number;
  height?: string | number;
  mt?: string | number;
  mb?: string | number;
  mr?: string | number;
  ml?: string | number;
}

const animation = keyframes`
  0% {
    opacity: 1 
  }
  50% {
    opacity: .2
  }
  100% {
    opacity: 1
  }
`;

const LineStyle: any = styled.div`
  width: ${({ width }: any) => width || 100}px;
  height: ${({ height }: any) => height || 10}px;
  margin-top: ${({ mt }: any) => mt || 10}px;
  margin-bottom: ${({ mb }: any) => mb || 10}px;
  margin-right: ${({ mr }: any) => mr || 10}px;
  margin-left: ${({ ml }: any) => ml || 10}px;
  background-color: #ddd;
  animation: ${animation} 0.8s linear;
  animation-iteration-count: infinite;
  animation-direction: normal;
  animation-fill-mode: forwards;
  border-radius: 4px;
`;

const Line: React.FC<Props> = ({ width, height, mb, mt, mr, ml, ...props }) => {
  return <LineStyle mb={mb} mt={mt} width={width} height={height} {...props} />;
};

export default Line;
