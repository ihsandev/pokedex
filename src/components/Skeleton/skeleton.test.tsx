import * as React from "react";
import { shallow } from "enzyme";
import Skeleton from "./index";

test("render Skeleton with props", () => {
  const wrapper = shallow(
    <Skeleton.Line width={40} height={40} mt={2} mb={2} mr={2} ml={2} />
  );
  expect(wrapper).toMatchSnapshot();
});
