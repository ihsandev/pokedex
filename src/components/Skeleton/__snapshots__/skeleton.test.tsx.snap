// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Skeleton with props 1`] = `
<styled.div
  height={40}
  mb={2}
  ml={2}
  mr={2}
  mt={2}
  width={40}
/>
`;
